rmc/utils/mpc.py

```diff
@@ -255,30 +257,72 @@ def create_context_with_oe(
     # before joining with RMC tables in `get_oe_annotation`
     # This computation is expensive, so overwrite only if specified (otherwise, read existing file)
     ht = ht.checkpoint(
-        f"{temp_path_with_del}/vep_context_mis_only_annot.ht",
+        f"{TEMP_PATH_WITH_SLOW_DEL}/vep_context_mis_only_annot.ht",
         _read_if_exists=not overwrite,
         overwrite=overwrite,
     )
 
     logger.info(
         "Adding regional missense constraint missense o/e annotation and writing to resource path..."
     )
-    ht = get_oe_annotation(ht)
-    ht = ht.checkpoint(
-        context_with_oe.path, _read_if_exists=not overwrite, overwrite=overwrite
-    )
+    ht = get_oe_annotation(ht, include_rescue)
+    ht = ht.key_by("locus", "alleles", "transcript")
+    oe_col = "oe_rescue" if include_rescue else "oe"
+    if include_rescue:
+        ht = ht.rename({"oe": oe_col})
+    # If `context_with_oe` HT already exists and overwrite is False,
+    # add column for newly-calculated OE
+    if file_exists(context_with_oe.path) and not overwrite:
+        context_with_oe_ht = context_with_oe.ht()
+        logger.info(
+            "OE-annotated context HT already exists with fields %s. Adding column %s to existing table...",
+            set(context_with_oe_ht.row),
+            oe_col,
+        )
+        ht = ht.select(oe_col)
+        ht = context_with_oe_ht.annotate(**ht[context_with_oe_ht.key])
```

Comment:
if these joins are slow, maybe we should add a checkpoint for ht prior to this line

Reply:
I think that's a good point, I moved the checkpoint in the next line up to here